packages/web-components/src/components/va-date/va-date.tsx

```diff
@@ -77,19 +82,31 @@ export class VaDate {
   })
   dateBlur: EventEmitter;
 
-  private handleDateBlur = (event: FocusEvent) => {
-    const [year, month, day] = (this.value || '')
-      .split('-')
-      .map(val => parseInt(val));
+  /**
+   * Set the value prop as an ISO-8601 date using provided arguments.
+   * Strips trailing hyphens and sets date to be null if the
+   * date values are all NaNs.
+   */
+  private setValue(year: number, month: number, day: number): void {
     // Use a leading zero for numbers < 10
     const numFormatter = new Intl.NumberFormat('en-US', {
       minimumIntegerDigits: 2,
     });
     // Ternary to prevent NaN displaying as value for Year
     // Ternary to prevent Month or Day from displaying as single digit
-    this.value = `${year ? year : ''}-${
+    const val = `${year ? year : ''}-${
       month ? numFormatter.format(month) : ''
-    }-${day ? numFormatter.format(day) : ''}`;
+    }-${day ? numFormatter.format(day) : ''}`.replace(/-+$/, '');
+
+    this.value = val ? val : null;
```

Comment:
Setting `value` to null when it would otherwise be an empty string just seemed more explicit.